src/ukrdc_cupid/core/match/identify.py

```diff
@@ -8,70 +8,227 @@
+    Will need to handle opt-outs eventually.
+
+    Args:
+        xml (xsd_ukrdc.Patient): 
+    """
+
+    # we can specify exactly which catagories we want to validate on later 
+    patient_info = {
+        "sending_extract" : xml.sending_extract.value,
+        "sending_facility" : xml.sending_facility.value,
+        "date_of_birth" : xml.patient.birth_time.to_datetime(),
+        #"gender" : xml.patient.gender,
+        #"postcodes" : [address.postcode for address in xml.patient.addresses.address],
+        "MRN" : None,
+        "NI" : []
+    }
+
+    # main matching against pid and ukrdc is done using patient numbers
+    for number in xml.patient.patient_numbers[0].patient_number:
+        if number.number_type.value == "MRN":
+            patient_info["MRN"] = [number.number, number.organization.value]
+
+        if number.number_type.value == "NI":
+            patient_info["NI"].append([number.number, number.organization.value]) 
+
+    # MRN is non negotiable for cupid matching
+    if not patient_info.get("MRN"):
+        raise Exception("placeholder exception...this should really never pass validation")
+
+    return patient_info
 
 
-def link_patient_number(session: Session, xml: xsd_ukrdc.PatientRecord):
-    """Look up patient from nhs number
+def identify_patient_feed(session: Session, patient_info:dict):
+    """Identify patient based on patient numbers.
 
     Args:
-        session (Session): ukrdc session
-        xml (xsd_ukrdc.PatientRecord): ukrdc patient record as xml
+        session (Session): SQLAlchemy session
+        xml (xsd_ukrdc.PatientRecord): UKRDC patient record as XML
     """
 
-    if xml.patient.patient_numbers:
-        """patient numbers are the universal source of truth we query these as a first step
-        if this is incorrect the patient will either be treated as new or matched to the
-        wrong patient. This is detectable by checking against demographics data. Do we want
-        to add this step in?
-        """
-
-        patient_ids = [
-            number.number for number in xml.patient.patient_numbers.patient_number
-        ]
-        patient_id_types = [
-            number.number_type.value
-            for number in xml.patient.patient_numbers.patient_number
-        ]
-
-        patient_number_query = (
-            select(orm.PatientRecord.pid, orm.PatientRecord.ukrdcid)
-            .join(orm.PatientNumber, orm.PatientNumber.pid == orm.PatientRecord.pid)
-            .filter(
-                orm.PatientRecord.sendingextract == xml.sending_extract.value,
-                orm.PatientRecord.sendingfacility == xml.sending_facility.value,
-                and_(
-                    orm.PatientNumber.patientid.in_(patient_ids),
-                    orm.PatientNumber.numbertype.in_(patient_id_types),
-                ),
-            )
+    # match patients on mrn
+    matched_patients_mrn = match_mrn(session, patient_info)
```

Comment:
George reminder to look at this more.

(My main question is around a changed MRN - but I need to study the flow chart).